src/include/mpir_request.h

```diff
@@ -301,6 +301,8 @@ static inline MPIR_Request *MPIR_Request_create_pool(MPIR_Request_kind_t kind, i
 {
     MPIR_Request *req;
 
+    /* hack for testing */
+    MPIDU_THREAD_ASSERT_IN_CS((*(MPID_Thread_mutex_t *) MPIR_Request_mem[pool].lock));
```

Comment:
This would be good to leave in for `--enable-g` builds, probably.

Reply:
Good idea. There are quite a few thread-related assertions that are really meant for debugging. Maybe we should add an option `--enable-g=thread`?

Reply:
we have these options already, though I do not know their state of functionality
```
        mutex    - Enable error checking on pthread mutexes
        mutexnesting - Check for non-nesting of mutexes
```

Reply:
Never aware of these 😄 I think we can combine the, both into `mutex`?

Reply:
I think so. `mutexnesting` doesn't appear to have any affect, anyway.

Reply:
Left the assertion inside `#ifdef MPICH_DEBUG_MUTEX`-guard. Also followed convention and renamed the macro `MPID_THREAD_ASSERT_IN_CS(VCI, mutex)`.